tests/scripting_test.cpp

```diff
@@ -45,23 +47,23 @@ BOOST_AUTO_TEST_CASE(TestGetProvider)
 
     PluginLoader::Instance()->loadService("scripting",tc);
 
-    // We use a sequential activity in order to force execution on trigger().
-    tc->stop();
-    BOOST_CHECK( tc->setActivity( new SequentialActivity() ) );
```

Comment:
All test cases in `scripting_test.cpp` still pass without switching to a true multi-threaded test using the default `Activity`, but some problematic effects of nested function calls and yield statements would be hidden if they would be executed sequentially by the main thread.